webserver/static/js/face.ts

```diff
@@ -0,0 +1,698 @@
+import {
```

Comment:
Not going to hold back from getting this merged, but I'd like to have a high-level overview of how the algorithm works at some stage added as a comment.